pkg/executables/kubectl.go

```diff
@@ -128,6 +128,7 @@ func (k *Kubectl) GetEksaCloudStackDatacenterConfig(ctx context.Context, cloudst
 	if err != nil {
 		return nil, fmt.Errorf("getting eksa cloudstack datacenterconfig: %v", err)
 	}
+	response.SetDefaults()
```

Comment:
Why are we doing this? IMO, this is a mistake, we shouldn't have these kind of logic at the client level.

We either set the defaults in a mutation webhook, or we set them from the cli when we create/update the object or we have some extra logic at the usecase level (maybe the provider?)

Reply:
How would we go about implementing a mutation webhook? The need for this call is when upgrading older clusters which are not using AvailabilityZones to the new AvailabilityZones schema, which is used internally in EKS-A throughout both the CLI and controller, as we are deprecating the old schema.

Reply:
When upgrading a cluster created from v1beta1 capc to eksa CLI with v1beta2 the retrieved cloudstack datacenter config is v1beta1 so we need conversion so I used SetDefaults() here in the original PR. If we think this is not a good place to call the function then we have to call it from the callers. What do you think?

Reply:
@maxdrib you can find mutation webhook for other CRDs: in the `_webhook.go` files in the api package

@wongni I'm ok calling this from the caller. I hope we only need to do it once. If we require this new format even in old objects, we should be able to pull, update and pull again.

Reply:
I don't know if mutation webhooks will work for our usecase. From what I can tell, mutation webhooks are intended to modify resources _before_ they're persisted in etcd. In our case, we have _existing_ resources in etcd from an older cluster which need to be converted to a new schema to be used in the EKS-A codebase, which can't seem to be done with mutating webhooks.